jupyter_server_proxy/__init__.py

```diff
@@ -28,8 +30,16 @@ def load_jupyter_server_extension(nbapp):
     server_handlers = make_handlers(base_url, server_proccesses)
     nbapp.web_app.add_handlers('.*', server_handlers)
 
+    # Configure SSL support
+    ssl_options = None
+    if serverproxy.https:
+        ssl_context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH, cafile=serverproxy.cafile)
+        ssl_context.load_cert_chain(serverproxy.certfile, serverproxy.keyfile)
```

Comment:
Is this a client certificate and key? If so could it be optional?

Reply:
I was intending that the `serverproxy.https` configurable would make it optional but perhaps you mean something else?  Thanks for taking a look!

Reply:
It's probably me misunderstanding how this works.

Is this so that jupyter-server-proxy can proxy a notebook URL such as `https://example.org/notebook/proxied-service/` through to a backend service that's only accessible over https? Or something else?

Reply:
The concrete case we are thinking of is where a Dask cluster is running on a HPC system and we turn on TLS to encrypt the traffic.  The Bokeh dashboard can also be configured to use TLS, and we'd like to proxy that dashboard through jupyter-server-proxy.

Reply:
Sorry for not following this up earlier.

For example you'd use it to proxy a site such as https://grafana.mybinder.org/ (though internal) via jupyter? This means:
1. jupyter-server-proxy connects to https://grafana.mybinder.org/. For SSL validation this requires the root CA public certificates (either bundled with Python if you're using an externally provided cert, or if it's an internal CA then configured in jupyter-server-proxy).
2. This is then passed through to the user's browser via jupyter notebook. If the notebook is protected by HTTPS (e.g. by Nginx) anything proxied by jupyter-server-proxy should also be protected since it all goes through the notebook server.

What I don't understand is where the requiremnt for a private key arises.